crates/rope/src/chunk.rs

```diff
@@ -536,21 +532,14 @@ impl Iterator for Tabs {
         let tab_offset = self.tabs.trailing_zeros() as usize;
         let chars_mask = (1 << tab_offset) - 1;
         let char_offset = (self.chars & chars_mask).count_ones() as usize;
-        self.byte_offset += tab_offset;
-        self.char_offset += char_offset;
+
+        // Since tabs are 1 byte the tab offset is the same as the byte offset
         let position = TabPosition {
-            byte_offset: self.byte_offset,
-            char_offset: self.char_offset,
+            byte_offset: tab_offset,
+            char_offset: char_offset
         };
-
-        self.byte_offset += 1;
-        self.char_offset += 1;
-        if self.byte_offset == MAX_BASE {
-            self.tabs = 0;
-        } else {
-            self.tabs >>= tab_offset + 1;
-            self.chars >>= tab_offset + 1;
-        }
+        // Remove the tab we've just seen
+        self.tabs ^= 1 << tab_offset;
```

Comment:
This could also be written
```suggestion
        self.tabs &= ~(1 << tab_offset);
```